### PR TITLE
Accessible Mark Read Indicator

### LINKED ISSUE #1448: Accessible post read status indicator
We had this in v1, it needs to be copied over to v2

### PR DESCRIPTION
**Pull request by EricBAndrews**
# Checklist
- [x] I have read CONTRIBUTING.md
- [x] I have described what this PR contains
- [x] If this PR alters the UI, I have attached pictures/videos
- [x] This PR addresses one or more open issues that were assigned to me:
      - closes #1448 

# Pull Request Information

Adds an accessible read post indicator. There are two options:
- Outline: unread posts are outlined. Thickness is configurable from 1 to 5px
- Checkmark: read posts show a checkmark next to the ellipsis menu

These markers should only be visible when the system-wide "differentiate without color" option is enabled.

### CODE REVIEW

File: Mlem/App/Views/Root/Tabs/Settings/AccessibilitySettingsView.swift [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+                        Slider(
+                            value: $readBarThicknessSlider,
+                            in: 1 ... 5,
+                            step: 1
+                        ) {
+                            Text("Outline Thickness")
+                        } minimumValueLabel: {
+                            Text("1")
+                        } maximumValueLabel: {
+                            Text("5")
```
Comment (Sjmarf) [inline Mlem/App/Views/Root/Tabs/Settings/AccessibilitySettingsView.swift]:
```suggestion
                            Text(verbatim: "1")
                        } maximumValueLabel: {
                            Text(verbatim: "5")
```

File: Mlem/Localizable.xcstrings [hunk 2]
```diff
@@ -96,6 +96,12 @@
     },
     "%lld on your instance" : {
 
+    },
+    "1" : {
+
+    },
+    "5" : {
+
```
Comment (Sjmarf) [inline Mlem/Localizable.xcstrings]:
```suggestion
```

**Review by Sjmarf — CHANGES_REQUESTED** [review]
Nice, the outline option works really well :D

I don't think we should have a "None" option in the accessibility setting. Did we have that in v1? Imo, if the user has "differentiate without color" enabled then we should accommodate that without requiring the user to change any settings manually, which they are currently required to do. Might defaulting to "checkmark" and removing the "none" option be better?

**Comment by EricBAndrews** [comment]
> if the user has "differentiate without color" enabled then we should accommodate that without requiring the user to change any settings manually, which they are currently required to do

Absolutely agreed, I've updated the default to checkmark. Nice catch!

> I don't think we should have a "None" option in the accessibility setting

I think we should keep that as an option because I don't want to make assumptions about why a user has the system-wide option enabled--e.g., a user may have it on because they are red/green colorblind but have no trouble differentiating `.primary` and `.secondary`, in which case they may not want the added visual clutter of the accessible indicator.

**Review by Sjmarf — APPROVED** [review]
Works for me 👍

**Pull request merged**